src/transformState.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    for (const key in actions[i]) {
+      if (actions[i][key] === 'addProperties') {
+        Object.assign(state, actions[i]['extraData']);
+      };
+
+      if (actions[i][key] === 'removeProperties') {
+        for (const key1 in actions[i]['keysToRemove']) {
```

Comment:
according to the checklist: 1. [CODE STYLE]: don't use for in loop for iterating over array